src/rpc/mining.cpp

```diff
@@ -585,177 +575,22 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
     }
 
     // Update block
```

Comment:
To simplify, we don't do template caching here anymore. I believe the cache invalidation was wrong anyway, as `setClientRules` was not respected. The end goal is *fast* `getblocktemplate` also.